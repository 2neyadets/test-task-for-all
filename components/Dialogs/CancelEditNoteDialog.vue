<template lang="pug">
  Dialog(:value="cancelEditNoteDialog" @hide="changeDialogState" @keyup.esc.native="changeDialogState")
    .delete-note-confirm
      h6.mt-none.mb-md(v-html="title" style="text-align: justify;")
      .flex.justify-end
        Btn(
          @click="changeDialogState"
          label="Нет, редактировать"
          size="16"
          flat
          autofocus
        )
        Btn.ml-sm(
          @click="cancelEditNote"
          label="Да, отменить"
          size="16"
        )
</template>

<script>
import { mapGetters } from 'vuex'
import Dialog from '../Reusable/Dialog'
import Btn from '../Reusable/Btn'

export default {
  name: 'CancelEditNoteDialog',
  components: { Btn, Dialog },
  props: {},
  data () {
    return {
    }
  },
  computed: {
    ...mapGetters([
      'cancelEditNoteDialog',
    ]),
    ...mapGetters('notes', [
      'currentNote',
    ]),
    title () {
      return `Вы уверены, что хотите <span style='color: red;'>отменить редактирование</span> заметки с названием <strong>"${this.currentNote && this.currentNote.title}"</strong>?`
    },
  },
  watch: {},
  created () {},
  mounted () {},
  updated () {},
  methods: {
    changeDialogState () {
      this.$store.dispatch('changeDialogState', 'cancelEditNote')
    },
    cancelEditNote () {
      this.changeDialogState()
      this.$router.push('/')
    },
  },
}
</script>
