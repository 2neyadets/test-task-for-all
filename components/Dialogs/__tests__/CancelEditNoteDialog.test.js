import Vuex from 'vuex'
import VueRouter from 'vue-router'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import CancelEditNoteDialog from '../CancelEditNoteDialog'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(VueRouter)
const router = new VueRouter()

describe('CancelEditNoteDialog component', () => {
  let wrapper, actions, noteGetters, getters, store
  const currentNote = {
    id: 'f1733253812c-f300272d',
    title: 'Название заметки',
  }
  router.push(`/note/${currentNote.id}`)

  beforeEach(() => {
    actions = {
      changeDialogState: jest.fn(),
      'notes/deleteNoteAction': jest.fn(),
    }
    noteGetters = {
      currentNote: () => ({ ...currentNote }),
    }
    getters = {
      cancelEditNoteDialog: () => true,
    }
    store = new Vuex.Store({
      actions,
      getters,
      modules: {
        notes: {
          namespaced: true,
          getters: noteGetters,
        },
      },
    })

    wrapper = shallowMount(CancelEditNoteDialog, {
      localVue,
      store,
      router,
    })
  })

  it('mounts without errors', done => {
    localVue.config.errorHandler = done // для отлавливания ошибок внутри nextTick, если нужно что-то выпонять внутри него
    expect.assertions(1)
    expect(wrapper).toBeTruthy()
    done()
  })

  it('can close dialog', () => {
    expect.assertions(1)
    const btn = wrapper.findAll('MyBtn-stub').at(0)
    btn.trigger('click')
    expect(actions.changeDialogState).toHaveBeenCalled()
  })

  it('can cancel edit note', () => {
    expect.assertions(2)
    const btn = wrapper.findAll('MyBtn-stub').at(1)
    btn.trigger('click')
    expect(actions.changeDialogState).toHaveBeenCalled()
    expect(router.currentRoute.fullPath).toBe('/')
  })

  it('contains note.title', () => {
    expect.assertions(1)
    expect(wrapper.find('H6').html()).toContain(currentNote.title)
  })

  it('match snapshot', () => {
    expect.assertions(1)
    expect(wrapper.element).toMatchSnapshot()
  })
})
