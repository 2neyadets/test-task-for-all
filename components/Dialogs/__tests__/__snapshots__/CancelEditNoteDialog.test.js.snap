// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelEditNoteDialog component match snapshot 1`] = `
<dialog-stub
  value="true"
>
  <div
    class="delete-note-confirm"
  >
    <h6
      class="mt-none mb-md"
      style="text-align: justify;"
    >
      Вы уверены, что хотите 
      <span
        style="color: red;"
      >
        отменить редактирование
      </span>
       заметки с названием 
      <strong>
        "Название заметки"
      </strong>
      ?
    </h6>
    <div
      class="flex justify-end"
    >
      <mybtn-stub
        autofocus="true"
        flat="true"
        icon=""
        label="Нет, редактировать"
        size="16"
        type="button"
      />
      <mybtn-stub
        class="ml-sm"
        icon=""
        label="Да, отменить"
        size="16"
        type="button"
      />
    </div>
  </div>
</dialog-stub>
`;
